srv/src/test/java/my/bookshop/AdminServiceTest.java

```diff
@@ -42,4 +48,21 @@ public void testUnauthorizedAccess() {
 		adminService.newDraft(Insert.into(Orders_.class).entry(Collections.emptyMap()));
 	}
 
+	@Test(expected = ServiceException.class)
+	@WithMockUser(username = "admin")
+	public void testInvalidAuthorName() {
+		Map<String, Object> data = new HashMap<>();
```

Comment:
You could use the typed interfaces here and in the other test method instead of the "generic" map:
```
Authors author = Authors.create()
author.setName("...");
```

Reply:
I changed to the typed interface.